Configure

```diff
@@ -151,6 +155,9 @@ my $clang_devteam_warn = ""
         . " -Wmissing-variable-declarations"
         ;
 
+# TODO!
+$devteam_additional_warn{icc} = "";
```

Comment:
Isn't this a bit like putting word in somebody's mouth? I mean "devteam" is us, right? But do we "have" icc? Or more specifically do we maintain icc? Well, Intel does offer free license to open-source developers (on per-year basis), but it also takes some will-power from us. Because by "maintain" I don't mean just pushing through --strict-warnings this time, but even arranging it so that build doesn't have dependencies on icc-specific run-time components and keeping it "in the loop"...

Actually with reference to Configurations/90-team.cf modification above, I'd actually suggest to create say Configurations/50-icc.cf, possibly with two targets, say linux-x86_64-icc and linux-x86_64-icc-strict-warnings with icc-specific flags. Or just linux-x86_64-icc alone but with warnings on, because one can as well use warnings in production build. This way one can simply omit this line in Configure.

Reply:
Alternatively `$developer_*_warn` would probably be better names...

Reply:
You should say that in #4248.

(re `$developer_*_warn`, why?  How does that make things better?)

Reply:
> $developer_*_warn, why?

It's not my first choice... As already said I'd rather suggest to have *nothing* icc-specific in Configure and limit it to Configurations/50-icc.cf.

Reply:
Yeah, but apart from that?  What's wrong with having a hash table keyed with compiler names, regardless of `icc`?  At some point, we might add another compiler among the set we (the team) decide to support, so we'd just add an entry...

The *only* reason I have *this* particular line is that `icc` is already part of `Configure` as it currently stands, see [here](https://github.com/openssl/openssl/commit/a610934c3bdf2c0aafc633d4245efe3df289d716).  The rest of this patch has nothing to do with `icc`.

Or do I misunderstand what you're talking about?

Reply:
All right. I think that adding icc as recognized option for --strict-warnings was *not* exactly justified. Why not? gcc and clang are multi-platform (both processor and os-wise) compilers *readily* available to *any* developer. icc on the other hand is none of those things. In a sense gcc and clang unite developers, icc divides, if you wish. Besides, as already suggested, I see no reason why the goal can't be achieved with custom dedicated separately-maintained config file. That's what they are meant to be used for. For niche things for somebody else to maintain. Not that I don't appreciate icc, or refuse to recognize the value of the exercise, but do we have to do *everything*?

Reply:
Nope, we don't have to do *everything*, and I wasn't my intention to say so either.  I just went with what was already there, with no intention to make more of a statement of it.

I haven't really focused on the config file idea, but I do think it's sound.  As a matter of fact, I've had some thoughts on moving compiler specific things to compiler configs that can be used as templates for other configs, and perhaps all of the strict-warnings related stuff should really end up there.

Reply:
> went with was already there

But it was put there just couple of days ago [and you were the committer]. You can't refer to it as if it was there forever...

Reply:
Look, it sounds like you would like a610934c3bdf2c0aafc633d4245efe3df289d716 to be reverted prior to this change, is that it?  *In practice*, that's what happens if I don't include this line.  If that's what you're after for now, *please say so*.

Reply:
>  If that's [removal of this line] what you're after for now, please say so.

But it says so in the very beginning of this thread. Just scroll up. It ends with "This way one can simply omit this line in Configure." Yes, it would effectively "incapacitate" a610934, but as already said I don't think it was really justified to start with. Whether or not it's handled by actual revert and then re-basing of this request doesn't really matter. Though if it's handled by simply omitting the line in this request, it might be appropriate to mention that it effectively obsoletes previous commit in commit message.

And you can always say "you're talking nonsense" and call for somebody else to concur :-) No problem with that either...

Reply:
How about following! Revert a610934, and *instead* collect output from `$config{cross_compile_prefix}$target{cc} -dM -E -x c /dev/null` and examine it for presence of __GNUC__, __clang__, whichever macros and base decisions on *that*. And I'm talking about *all* compiler-specific decisions, not just this --strict-warnings one. This would actually allow to resolve several small and subtle problems. But would it help icc you might ask. Yes, [recent] icc [just like few other vendor compilers] actually defines __GNUC__, presumably to facilitate portability. As as far as required -Wno-whatever-s that suits icc go, they can be specified at Configure command line or maintained in Confgirations/50-icc.cf [when/if contributed].

Reply:
This sounds like a solid foundation to build upon.  It's got my vote.

Reply:
See #4281.